tests/form-filling.js

```diff
@@ -0,0 +1,101 @@
+const helper = require('./helper');
+const genericFormPo = require('./page-objects/forms/generic-form.po');
+const answeredQuestions = [];
+
+const getPaths = (obj, prevKey, r = []) => {
+  const results = r;
+  let evalObj = obj;
+  if (Array.isArray(obj)) {
+    evalObj = obj[0];
+  }
+  Object.keys(evalObj).forEach((key) => {
+    let resultKey = '';
+    if(prevKey){
+      resultKey = `${prevKey}/${key}`;
+    } else {
+      resultKey = key;
+    }
+
+    if(key === 'inputs'){
+      return;
+    }
+    if(typeof evalObj[key] !== 'object'){
+      results.push({path: resultKey, value: evalObj[key]});
+    }else {
+      getPaths(evalObj[key], resultKey, results);
+    }
+  });
+  return results;
+};
+
+
+const getVisibleQuestions = async () => {
+  const answers = await element.all(by.css('.current input:not([disabled],.ignore)'));
+  const results = [];
+
+  for(let i = 0; i < answers.length; i++){
+    let name = await answers[i].getAttribute('data-name');
+    let cssAttribute = 'data-name';
+    if(!name) {
+      name = await answers[i].getAttribute('name');
+      cssAttribute = 'name';
+    }
+    const type = await answers[i].getAttribute('type');
+    if (results.filter(e => e.name === name && e.type === type).length === 0) {
```

Comment:
Instead of `filter` and a length check, it is preferred to use [some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some). Most importantly, it's clearer what you're doing, but also there's a performance benefit (not important here really though) that it'll stop looping as soon as it finds a match.

```suggestion
    if (!results.some(e => e.name === name && e.type === type)) {
```